src/NUnit.Portable.Agent/NUnitPortableDriver.cs

```diff
@@ -69,17 +65,15 @@ public class NUnitPortableDriver
         public string Load(Assembly frameworkAssembly, Assembly testAssembly, IDictionary<string, object> settings)
         {
             var idPrefix = string.IsNullOrEmpty(ID) ? "" : ID + "-";
-            _frameworkAssembly = frameworkAssembly;
-            _testAssembly = testAssembly;
-
-            _frameworkController = CreateObject(CONTROLLER_TYPE, testAssembly, idPrefix, settings);
-            if (_frameworkController == null)
+            var frameworkController = CreateObject(CONTROLLER_TYPE, frameworkAssembly, testAssembly, idPrefix, settings);
+            if (frameworkController == null)
                 throw new NUnitPortableDriverException(INVALID_FRAMEWORK_MESSAGE);
 
-            _frameworkControllerType = _frameworkController.GetType();
+            var testPackage = new TestPackage(testAssembly, frameworkAssembly, frameworkController);
+            _testPackages.Add(testPackage);
 
-            log.Info("Loading {0} - see separate log file", _testAssembly.FullName);
-            return ExecuteMethod(LOAD_METHOD) as string;
+            log.Info("Loading {0} - see separate log file", testAssembly.FullName);
+            return testPackage.ExecuteMethod(LOAD_METHOD) as string;
         }
```

Comment:
The intention was that runners will still call the agent's `Load()` method per assembly. In this case, what should `Load()` return, the XML result from the single assembly you've just loaded, or XML for all the assemblies loaded so far?

I had a look around - the result from this isn't used in dotnet-test-nunit, and the console abstracts at a different level, so it loads all the assemblies together, and then returns a single XML result. What would a result from `Load()` actually be used for? I'm not sure of the best way to go here.